File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // unordered_map<int, int> map;

        // O(n^2) time
        // O(1) space
        vector<int> solution;

        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    solution.push_back(i);
                    solution.push_back(j);
                }
            }
        }
        
        return solution;
    }
};  

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hm = {}

        for index, num in enumerate(nums):
            if target - num in hm:
                return [hm[target - num], index]
            else:
                hm[num] = index